scene/gui/code_edit.cpp

```diff
@@ -2415,6 +2421,19 @@ void CodeEdit::confirm_code_completion(bool p_replace) {
 				}
 			}
 		}
+
+		for (const ScriptLanguage::TextEdit &edit : selected_option.additional_edits) {
+			ScriptLanguage::CodePos start = edit.start;
+			ScriptLanguage::CodePos end = edit.end;
+
+			int nb_lines = this->get_text().size();
+
+			this->remove_text(start.line, start.column, end.line, end.column);
+			this->insert_text(edit.new_text, start.line, start.column);
+
+			int new_nb_lines = this->get_text().size();
+			// TODO: Does the caret magically move or do we need to update it?
+		}
```

Comment:
Another mental TODO here is that this currently applies each time for a multi-caret completion.

Given that the positions in `ScriptLanguage::TextEdit` are absolute, and are not aware of the fact that a completion could be used for multiple carets, we might want to only apply them once after all carets have completed

https://i.imgur.com/9liZOSc.mp4